cli/plasmo/src/features/manifest-factory/base.ts

```diff
@@ -419,17 +420,47 @@ export abstract class PlasmoManifest<T extends ExtensionManifest = any> {
       .then((files) =>
         Promise.all(
           files
-            .filter((f) => f.isFile())
+            .filter((f) =>
+              f.isFile() && filterFile ? filterFile(f.name) : true
+            )
             .map((f) => resolve(path, f.name))
             .map((filePath) => toggleDynamicPath(filePath, true))
         )
       )
       .then((results) => results.includes(true))
   }
 
+  addContentScriptsNestedDirectories = async () => {
+    const path = this.projectPath.contentsDirectory
+
+    const indexFileList = [...this.#extSet].flatMap((ext) => [
+      `index${ext}`,
+      `index.${this.browser}${ext}`
+    ])
```

Comment:
This is more or less pulled from [`getIndexList`](https://github.com/PlasmoHQ/plasmo/blob/1872f6303bf61d36c553badc6cec80dfb2728de0/cli/plasmo/src/features/extension-devtools/project-path.ts#L66), minus the `NODE_ENV` logic which afaik isn't applicable here